common/hyperscaler/rules/parsing_result.go

```diff
@@ -0,0 +1,47 @@
+package rules
+
+import "github.com/google/uuid"
+
+type ParsingResult struct {
+	ID uuid.UUID
+
+	OriginalRule string
+
+	Rule *Rule
+	// array with errors that occurred during parsing of rule entry
+	ParsingErrors []error
+
+	// array with errors that occurred after successful rule parsing
```

Comment:
What kind of errors can occur after parsing?
Duplication?
Inconsistency?

Reply:
Uniqueness validation errors and signature check ("mirror" logic).

Reply:
So could we name it SignatureAndUniquenessErrors?
Or similarly.
Parsing is processing as well, so we would have clear distingtion:

- ParsingErrors,
- SignatureAndUniquenessErrors.

Reply:
Maybe in the future you would need to add another validation, e.g. dictionary check, would you extend this name then and call the method `SignatureAndUniquenessAndDictionaryErrors`? Now the method refers to the stage in rules parsing processing.

Reply:
No. I would consider adding DictionaryErrors.
Or keep single collection with attribute determining to which stage these relate.